cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -217,7 +217,9 @@ public void create(String namespaceId, String name, DatasetInstanceConfiguration
     DatasetSpecification spec = opExecutorClient.create(newInstance, typeMeta,
                                                         DatasetProperties.builder()
                                                           .addAll(props.getProperties())
-                                                          .build());
```

Comment:
if the builder had a way to add a description, you could use that here alternatively, But the configure() method would also be able to add one.